src/log/log.c

```diff
@@ -2387,6 +2400,13 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
             if (LF_ISSET(WT_LOGSCAN_ONE))
                 break;
         }
+
+        if (end_range_lsnp != NULL && next_lsn.l.file > end_range_lsnp->l.file) {
```

Comment:
I would change this to:
```
If (end_range_lsnp != NULL && (next_lsn.l.file > end_range_lsnp->l.file || next_lsn.l.file == end_range_lsnp->l.file &&
          next_lsn.l.offset > end_range_lsnp->l.offset)) {
```
That way we have one less if, additionally a comment explaining what's happening would be good because it will be a complex if.

Reply:
Yeah I avoided doing that for readability sake. Is it more efficient to do it in the one statement?

Reply:
It's not really a question of efficiency, but code compactness, I think one could argue either way for readability. That said having two if's which both result in the same outcome would be confusing to me as a reader.

Reply:
Yeah that's fair enough so is compactness generally prioritised over readability?

Reply:
No, I don't think so I think its case specific however I wouldn't say the if statement duplication improved readability.

Reply:
Still would like a comment explaining that so a reader doesn't need to spend long parsing the if.